### PR TITLE
add postbuild targets

### PR DESCRIPTION
**Pull request by MS-TimothyMothra**
Fix: https://github.com/Microsoft/ApplicationInsights-dotnet/issues/1021

### CODE REVIEW

**Comment by pharring** [comment]
Have you tried using a .props file (instead of .targets) and setting the property there instead?

**Comment by MS-TimothyMothra** [comment]
@pharring I didn't realize that property would be accessible in a .props file.
I found an example [here](https://stackoverflow.com/questions/21143817/set-content-files-to-copy-local-always-in-a-nuget-package/30386836#30386836). I'll test it this afternoon.

File: Src/DependencyCollector/DependencyCollector/Microsoft.ApplicationInsights.DependencyCollector.targets [hunk 1]
```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="utf-8"?>
+<Project ToolsVersion="4.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+
+  <Target Name="CopyApplicationInsightsConfigToOutputDirectory" AfterTargets="CopyFilesToOutputDirectory" >
```
Comment (SergeyKanzhelev) [inline Src/DependencyCollector/DependencyCollector/Microsoft.ApplicationInsights.DependencyCollector.targets]: can you please set affected files as a property of this target

File: Src/DependencyCollector/DependencyCollector/Microsoft.ApplicationInsights.DependencyCollector.targets [hunk 2]
```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="utf-8"?>
+<Project ToolsVersion="4.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
```
Comment (SergeyKanzhelev) [inline Src/DependencyCollector/DependencyCollector/Microsoft.ApplicationInsights.DependencyCollector.targets]: are we forced to specify ToolsVersion here?

File: Src/DependencyCollector/DependencyCollector/Microsoft.ApplicationInsights.DependencyCollector.targets (hunk 1 above)
Comment (SergeyKanzhelev) [inline Src/DependencyCollector/DependencyCollector/Microsoft.ApplicationInsights.DependencyCollector.targets]: Is `CopyFilesToOutputDirectory` executes for console apps?

Comment (MS-TimothyMothra) [reply Src/DependencyCollector/DependencyCollector/Microsoft.ApplicationInsights.DependencyCollector.targets]: sorry, i don't understand what you're asking here

Comment (MS-TimothyMothra) [reply Src/DependencyCollector/DependencyCollector/Microsoft.ApplicationInsights.DependencyCollector.targets]: I can test specifically with console apps. Would we need to take any action if `CopyFilesToOutputDirectory` is not executed?

Comment (MS-TimothyMothra) [reply Src/DependencyCollector/DependencyCollector/Microsoft.ApplicationInsights.DependencyCollector.targets]: I confirmed that this is executed for Console apps.

I would argue though that we would be unaffected by any app type that does not copy files to an output directory. :)

File: Src/DependencyCollector/DependencyCollector/Microsoft.ApplicationInsights.DependencyCollector.targets (hunk 2 above)
Comment (MS-TimothyMothra) [reply Src/DependencyCollector/DependencyCollector/Microsoft.ApplicationInsights.DependencyCollector.targets]: removing

**Comment by MS-TimothyMothra** [comment]
@pharring I haven't been able to find a setting that modifies the file's properties. 
If you know of one, please do share!

MSBuild Tasks can let me execute scripts to do this, but I would need time to investigate a script. 
I'm not sure if this is worth the time right now.

**Comment by pharring** [comment]
Does the following work?
```xml
<?xml version="1.0" encoding="utf-8"?>
<Project ToolsVersion="15.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
  <ItemGroup>
    <Content Include="ApplicationInsights.config">
    <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
    </Content>
  </ItemGroup>
</Project>
```

**Pull request merged**